src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -639,6 +633,53 @@ RemoteTransactionsBeginIfNecessary(List *connectionList)
 }
 
 
+/*
```

Comment:
:confused: there's a _lot_ of similarity in these two methods and I worry about them diverging, but I suppose this is OK (apparently `ReportConnectionError` and `ReportResultError` were already broken apart in the past?)

Reply:
There is, but `ReportResultError` is wildly different from `ReportConnectionError` and takes a `PGresult *`.